mutiny-core/src/ldkstorage.rs

```diff
@@ -88,26 +92,55 @@ impl<S: MutinyStorage> MutinyNodePersister<S> {
         format!("{}_{}", key, self.node_id)
     }
 
-    fn persist_local_storage<W: Writeable>(
+    fn init_persist_local_storage<W: Writeable>(
         &self,
-        key: &str,
+        key: String,
         object: &W,
-        version: Option<u32>,
-    ) -> Result<(), lightning::io::Error> {
-        let key_with_node = self.get_key(key);
-        self.storage
-            .set_data(key_with_node, object.encode(), version)
-            .map_err(|e| {
-                match e {
-                    MutinyError::PersistenceFailed { source } => {
-                        log_error!(self.logger, "Persistence failed on {key}: {source}");
+        version: u32,
+        update_id: MonitorUpdateIdentifier,
+    ) -> ChannelMonitorUpdateStatus {
+        let storage = self.storage.clone();
+        let chain_monitor = self.chain_monitor.clone();
+        let logger = self.logger.clone();
+        let object = object.encode();
+        spawn(async move {
```

Comment:
Why is `init_persist_local_storage` still saving in a spawn? I see `ChannelMonitorUpdateStatus::InProgress` is being returned but where is it ever told that it completed?

Reply:
This function still needs to be synchronous the write is just async. We return `ChannelMonitorUpdateStatus::InProgress` here so ldk knows that is happening in the background. In `persist_local_storage` we call `chain_monitor.channel_monitor_updated` which marks it complete